src/codeu/chat/client/core/ConversationContext.java

```diff
@@ -115,4 +115,29 @@ private MessageContext getMessage(Uuid id) {
     return messages.hasNext() ? new MessageContext(messages.next(), view) : null;
   }
 
-}
+  /**
+   * Adds a bot the conversationHeader associated with this context
+   * @param botName name of the bot being added
+   * @return
+   */
+  public boolean addBot(String botName) {	
+	if (controller.addBot(this.conversation.id, botName)) {
+	  this.conversation.bots.add(botName);
+	  return true;
+	}
+	return false;	
+  }
+
+  /**
+   * removes bot from conversationheader associated with this context
+   * @param botName name of bot being removed
+   * @return
+   */
+  public boolean removeBot(String botName) {
+	if (controller.removeBot(this.conversation.id, botName)) {
+      this.conversation.bots.remove(botName);
```

Comment:
I'm not sure if this will work, because the list on conversation is now of type Bot, not string